_includes/sidebar.html

```diff
@@ -1,91 +1,34 @@
-      <li><a href="{{ '/kit/servo_board' | prepend: site.baseurl }}">Servo Board</a></li>
-      <li><a href="{{ '/kit/wifi' | prepend: site.baseurl }}">WiFi</a></li>
-    </ul>
-  </li>
-  <li>
-    <a href="{{ '/programming/' | prepend: site.baseurl }}">Programming</a>
-    <ul>
-      <li>
-        <a href="{{ '/programming/python/' | prepend: site.baseurl }}">Python</a>
-        <ul>
-          <li><a href="{{ '/programming/python/functions' | prepend: site.baseurl }}">Functions</a></li>
-          <li><a href="{{ '/programming/python/libraries' | prepend: site.baseurl }}">Libraries</a></li>
-        </ul>
-      </li>
-      <li>
-        <a href="{{ '/programming/sr/' | prepend: site.baseurl }}">sr</a>
-        <ul>
-          <li><a href="{{ '/programming/sr/motors/' | prepend: site.baseurl }}">Motors</a></li>
-          <li><a href="{{ '/programming/sr/power/' | prepend: site.baseurl }}">Power</a></li>
-          <li>
-            <a href="{{ '/programming/sr/ruggeduinos/' | prepend: site.baseurl }}">Ruggeduinos</a>
-            <ul>
-              <li><a href="{{ '/programming/sr/ruggeduinos/custom_firmware' | prepend: site.baseurl }}">Custom Firmware</a></li>
-            </ul>
-          </li>
-          <li><a href="{{ '/programming/sr/servos/' | prepend: site.baseurl }}">Servos</a></li>
-          <li>
-            <a href="{{ '/programming/sr/vision/' | prepend: site.baseurl }}">Vision</a>
-            <ul>
-              <li><a href="{{ '/programming/sr/vision/markers' | prepend: site.baseurl }}">Markers</a></li>
-            </ul>
-          </li>
-        </ul>
-      </li>
-      <li><a href="{{ '/programming/git_repositories' | prepend: site.baseurl }}">Git Repositories</a></li>
-      <li><a href="{{ '/programming/simulator' | prepend: site.baseurl }}">Simulator</a></li>
-    </ul>
-  </li>
-  <li><a href="{{ '/rules/' | prepend: site.baseurl }}">Rules</a></li>
-  <li>
-    <a href="{{ '/troubleshooting/' | prepend: site.baseurl }}">Troubleshooting</a>
-    <ul>
-      <li><a href="{{ '/troubleshooting/python' | prepend: site.baseurl }}">Python</a></li>
-      <li><a href="{{ '/troubleshooting/interactive_troubleshooter' | prepend: site.baseurl }}">Interactive Troubleshooter</a></li>
-    </ul>
-  </li>
-  <li>
-    <a href="{{ '/tutorials/' | prepend: site.baseurl }}">Tutorials</a>
-    <ul>
-      <li><a href="{{ '/tutorials/basic_motor_control' | prepend: site.baseurl }}">Basic Motor Control</a></li>
-      <li><a href="{{ '/tutorials/python' | prepend: site.baseurl }}">Python</a></li>
-    </ul>
-  </li>
-  <li>
-    <a href="{{ '/team_admin/' | prepend: site.baseurl }}">Team Admin</a>
-    <ul>
-      <li><a href="{{ '/team_admin/user_accounts' | prepend: site.baseurl }}">User Accounts</a></li>
-      <li><a href="{{ '/team_admin/kit_shipping' | prepend: site.baseurl }}">Kit Shipping</a></li>
-    </ul>
+  {% for heading in site.data.sidebar_tree.tree %}
```

Comment:
Yeah I've hit the limitations of Liquid many times while writing this

Reply:
Hrm, I wonder whether a further improvement (for another time) would be to investigate or create a Jekyll plugin for this rather than doing it in Liquid?

This is already a big step up from what we had though :)